python/ray/train/tensorflow/tensorflow_checkpoint.py

```diff
@@ -48,9 +73,175 @@ def from_model(
+        """
+        assert path.isfile(file_path) and file_path.endswith(
+            ".h5"
+        ), "Please supply a h5 file path to `TensorflowCheckpoint.from_h5()`."
+        dir_path = path.dirname(os.path.abspath(file_path))
+        if preprocessor:
+            save_preprocessor_to_dir(preprocessor, dir_path)
+        checkpoint = cls.from_directory(dir_path)
+        checkpoint._flavor = cls.Flavor.H5
+        return checkpoint
+
+    @classmethod
+    def from_saved_model(
+        cls, dir_path: str, *, preprocessor: Optional["Preprocessor"] = None
+    ) -> "TensorflowCheckpoint":
+        """Create a :py:class:`~ray.air.checkpoint.Checkpoint` that stores a Keras
+        model - from SavedModel format.
+
+        The path must maintain valid even after this function returns.
+
+        Args:
+            path: The same path as used by ``model.save(path)``.
+            preprocessor: A fitted preprocessor to be applied before inference.
+
+        Returns:
+            A :py:class:`TensorflowCheckpoint` converted from SavedModel format.
+
+        Examples:
+
+            >>> import tensorflow as tf
+
+            >>> import ray
+            >>> from ray.train.batch_predictor import BatchPredictor
+            >>> from ray.train.tensorflow import (
+            ... TensorflowCheckpoint, TensorflowTrainer, TensorflowPredictor)
+            >>> from ray.air import session
+            >>> from ray.air.config import ScalingConfig
+
+            >>> def train_fn():
+            ...     model = tf.keras.Sequential(
+            ...         [
+            ...             tf.keras.layers.InputLayer(input_shape=()),
+            ...             tf.keras.layers.Flatten(),
+            ...             tf.keras.layers.Dense(10),
+            ...             tf.keras.layers.Dense(1),
+            ...         ])
+            ...     model.save("my_model")
+            ...     checkpoint = TensorflowCheckpoint.from_saved_model("my_model")
+            ...     session.report({"my_metric": 1}, checkpoint=checkpoint)
+
+            >>> trainer = TensorflowTrainer(
+            ...     train_loop_per_worker=train_fn,
+            ...     scaling_config=ScalingConfig(num_workers=2))
+
+            >>> result_checkpoint = trainer.fit().checkpoint
+
+            >>> batch_predictor = BatchPredictor.from_checkpoint(
+            ...     result_checkpoint, TensorflowPredictor)
+            >>> batch_predictor.predict(ray.data.range(3))
+        """
+        if preprocessor:
+            save_preprocessor_to_dir(preprocessor, dir_path)
+        checkpoint = cls.from_directory(dir_path)
+        checkpoint._flavor = cls.Flavor.SAVED_MODEL
         return checkpoint
 
-    def get_model_weights(self) -> tf.keras.Model:
-        """Retrieve the model weights stored in this checkpoint."""
-        model_weights, _ = _load_checkpoint(self, "TensorflowTrainer")
-        return model_weights
+    def get_model_and_preprocessor(
```

Comment:
Can we have this be just `get_model` only? There is already a `get_preprocessor` defined in the superclass. `get_model` will keep this API consistent with TorchCheckpoint for example

Reply:
Also, I think we may need to add some abstraction like this

```python
@DeveloperAPI
def ModelCheckpoint(Checkpoint):
    
    @abc.abstractmethod
    def get_model(self):
        raise NotImplementedError
```

That way the Predictors can reliably call on the `get_model` API since it is guaranteed to be implemented.